OpenRA.Mods.Common/Traits/Render/CustomTerrainDebugOverlay.cs

```diff
@@ -0,0 +1,84 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Commands;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Displays custom terrain types.")]
+	class CustomTerrainDebugOverlayInfo : ITraitInfo
+	{
+		public readonly string Font = "TinyBold";
+
+		public object Create(ActorInitializer init) { return new CustomTerrainDebugOverlay(init.Self, this); }
+	}
+
+	class CustomTerrainDebugOverlay : IWorldLoaded, IChatCommand, IRender
+	{
+		const string CommandName = "debugcustomterrain";
+		const string CommandDesc = "Toggles the custom terrain debug overlay. Optional parameter: 'allcells'";
```

Comment:
Adding the changes requested tag to either remove this or explain the use case that justifies it.